targets/TARGET_STM/hal_tick_32b.c

```diff
@@ -44,8 +42,6 @@ void timer_irq_handler(void)
             __HAL_TIM_CLEAR_IT(&TimMasterHandle, TIM_IT_CC2);
```

Comment:
I like the idea - great PR ! 
now if the HAL_GetTick relies on common ticker layer, then I think that the complete code related to the usage of TIM_IT_CC2 can be removed I guess - don't you think ? 
This would mean no need to enable/disable CC2 IT, to need to manage the interrupt anymore....

Reply:
Yes I think you're right. The timer channel2 was here only to generate an interrupt for the HAL tick. I'll make a try.

Reply:
It works: I have removed all the code related to timer channel 2 and all mbedOS5 timer tests are PASS. Tested with both 16-bits and 32-bits timers.